rclcpp/test/rclcpp/test_intra_process_manager.cpp

```diff
@@ -820,3 +887,82 @@ TEST(TestIntraProcessManager, lowest_available_capacity) {
   c1 = ipm->lowest_available_capacity(p1_id);
   ASSERT_EQ(history_depth - 1u, c1);
 }
+
+/*
+   This tests the check inside add_publisher for transient_local
+   durability publishers
+   - add_publisher should throw runtime_error when no valid buffer ptr
+   is passed with a transient_local publisher
+ */
```

Comment:
here and else where.

```suggestion
/*
 * This tests the check inside add_publisher for transient_local
 * durability publishers
 * - add_publisher should throw runtime_error when no valid buffer ptr
 * is passed with a transient_local publisher
 */
```